proxy/grpcproxy/watch_client_adapter.go

```diff
@@ -73,17 +75,34 @@ type chanServerStream struct {
 	headerc  chan<- metadata.MD
 	trailerc chan<- metadata.MD
 	grpc.Stream
+
+	headers []metadata.MD
 }
 
 func (ss *chanServerStream) SendHeader(md metadata.MD) error {
+	outmd := make(map[string][]string)
+	for _, h := range append(ss.headers, md) {
+		for k, v := range h {
+			outmd[k] = v
+		}
+	}
 	select {
-	case ss.headerc <- md:
+	case ss.headerc <- outmd:
+		ss.headerc = nil
```

Comment:
oh, yeah it should probably return an error; it should only be called once according to the docs